### PR TITLE
Add JAX Infrequently Asked Questions (IAQ)

### PR DESCRIPTION
**Pull request by 8bitmp3**
Add JAX Infrequently Asked Questions (IAQ) - useful questions with answers that do not fall into the FAQ category (yet).

Item 1: How to apply `jax.device_put` on a `jax.Array` that has different values per host into a 'data' mesh axis?

### CODE REVIEW

**Comment by 8bitmp3** [comment]
@jakevdp This can be WIP or a good first PR. LMKWYT.

File: docs/iaq.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
+JAX Infrequently Asked Questions (IAQ)
+======================================
+
+.. _JAX - Frequently Asked Questions (FAQ): https://jax.readthedocs.io/en/latest/faq.html
+.. _JAX - The Sharp Bits: https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html
+
+This document represents a collection of *infrequently asked questions (IAQ)*. Contributions welcome!
+
+How to apply `jax.device_put` on an `jax.Array`` that has different values per host into a 'data' mesh axis?
```
Comment (8bitmp3) [inline docs/iaq.rst:9]:
```suggestion
How to apply `jax.device_put` on a `jax.Array` that has different values per host into a 'data' mesh axis?
```

File: docs/iaq.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
+JAX Infrequently Asked Questions (IAQ)
+======================================
+
+.. _JAX - Frequently Asked Questions (FAQ): https://jax.readthedocs.io/en/latest/faq.html
+.. _JAX - The Sharp Bits: https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html
+
+This document represents a collection of *infrequently asked questions (IAQ)*. Contributions welcome!
+
+How to apply ``jax.device_put`` on a ``jax.Array` that has different values per host into a 'data' mesh axis?
```
Comment (yashk2810) [inline docs/iaq.rst:9]: what does this mean?

Please don't copy internal code into public docs too (as you did below)

Comment (8bitmp3) [reply docs/iaq.rst:9]: This is from https://github.com/google/jax/blob/main/jax/_src/array.py#647

Comment (8bitmp3) [reply docs/iaq.rst:9]: Better link (permalink from GitHub) https://github.com/google/jax/blob/d03187e3054fe95c57e0b611ba2e399a9ce168fe/jax/_src/array.py#L647

Comment (yashk2810) [reply docs/iaq.rst:9]: Let's back up a bit.. what does that question mean?

Why do you need a different doc for it? I don't think this question belongs in this doc.

File: docs/iaq.rst [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+------------------------------------------------------------------------------------------------------------
+
+Here's an example from :meth:`jax.Array` (Line 647 <https://github.com/google/jax/blob/main/jax/_src/array.py#647>`_):
```
Comment (8bitmp3) [inline docs/iaq.rst:12]:
```suggestion
Here's an example from :meth:`jax.Array` (Line 647 <https://github.com/google/jax/blob/d03187e3054fe95c57e0b611ba2e399a9ce168fe/jax/_src/array.py#L647>`_):
```

UPDATE: Fixed the GitHub permalink

**Comment by yashk2810** [comment]
I don't think this question belongs in this doc FWIW. It should exist in a different doc which we have plans to write (someday).

**Comment by 8bitmp3** [comment]
> I don't think this question belongs in this doc FWIW. It should exist in a different doc which we have plans to write (someday).

:+1:  Will close this for now. Thanks!
